benchmark/README.rst

```diff
@@ -31,6 +31,18 @@ The script has three main functions.
   The performance plot shows the execution time of the benchmark as a function of a scalar
   "size parameter" ``n``, the exact meaning of which depends on each benchmark.
 
+Comparing Revisions
```

Comment:
How are the `.rst` files from `benchmark` included for the documentation? Perhaps we might have to revisit this topic (and might potentially be out of scope here).

Reply:
Currently, they are designed just to show up on GitHub, see https://github.com/XanaduAI/pennylane/tree/master/benchmark/.

This is very similar to how PyTorch and a lot of other packages do it: https://github.com/pytorch/pytorch/tree/master/benchmarks

If we decide to package this directory with the main pennylane package, then we can think about including this in the existing documentation. However, that's a decision for the future!

Reply:
Oh great, thanks @josh146! :blush:

Reply:
What about making this a separate repository? I found it quite annoying that when I wrote a new benchmark (on a separate branch) I had to switch branches all the time if I wanted to benchmark a PR.

Reply:
Could work potentially! A hacky way of handling that right now would be merging in the other PR branch into the benchmark one locally and then resetting after that. But this is pretty error-prone altogether.

Reply:
> What about making this a separate repository? I found it quite annoying that when I wrote a new benchmark (on a separate branch) I had to switch branches all the time if I wanted to benchmark a PR.

This is definitely an option, this is what tensorflow does: https://github.com/tensorflow/benchmarks